js-envs-test-kit/src/main/scala/org/scalajs/jsenv/test/kit/TestKit.scala

```diff
@@ -9,22 +9,64 @@ import java.util.concurrent.Executors
 import org.scalajs.io.MemVirtualBinaryFile
 import org.scalajs.jsenv._
 
+/** TestKit is a utility class to simplify testing of [[JSEnv]]s
+ *
+ *  It is mostly used by Scala.js' provided [[JSEnv]] test suite but it may be
+ *  used for additional tests specific to a particular [[JSEnv]].
+ *
+ *  @example
+ *  {{{
+ *  import scala.concurrent.duration._
+ *
+ *  val kit = new TestKit(new MyEnv, 1.second)
+ *  kit.withRun("""console.log("Hello World");""") {
+ *    _.expectOut("Hello World\n")
+ *      .closeRun()
+ *  }
+ *  }}}
+ *
+ *  @note Methods in [[TestKit]] allow to take a string instead of an [[Input]].
+ *      The string is converted into an input form supported by the [[JSEnv]] to
```

Comment:
That phrasing is weird. It suggests that `TestKit` will automagically decide how best to convert the string into an `Input`, depending on what the given `JSEnv` supports. What if in the future we have another kind of `Input`, and a `JSEnv` does not support `ScriptsToLoad`?

I suggest directly saying that a string corresponds to an `Input.ScriptsToLoad` with the string as only script.

Reply:
This is precisely what this phrasing is supposed to mean. We will need something like this to have a portable test suite. It is the reason why I backed down from having an implicit conversion `String => Input`. Instead, the current API will allow us to do something like:

```scala
def tryStart(input: Input): Option[Run] = {
  try {
    Some(start(input))
  } catch {
    case _: UnsupportedInputException => None
}

tryStart(Input.ScriptsToLoad(...)) orElse tryStart(Input.OtherType(...))
```

Reply:
Ah, I see. OK.